treeherder/model/derived/jobs.py

```diff
@@ -347,7 +349,8 @@ def cancel_job(self, requester, job):
                            project_specific_id=job['id'],
                            state='pending').update(
                                state='completed',
-                               result='usercancel')
+                               result='usercancel',
+                               last_modified=datetime.now())
```

Comment:
On second thought, I'm not sure if that's such a good idea in most cases as it just makes the code a bunch more verbose.

There's a lot of awkwardness here that comes from the fact that we're still writing to the per-project datasource tables. I promise to clean up this code when I drop that.